internal/services/springcloud/spring_cloud_build_deployment_resource.go

```diff
@@ -83,30 +83,16 @@ func resourceSpringCloudBuildDeployment() *pluginsdk.Resource {
 							Type:     pluginsdk.TypeString,
 							Optional: true,
 							Computed: true,
-							ValidateFunc: validation.StringInSlice([]string{
-								"500m",
-								"1",
-								"2",
-								"3",
-								"4",
-							}, false),
+							// There are additional options when the service team allow them
+							ValidateFunc: validation.StringIsNotEmpty,
 						},
 
 						"memory": {
 							Type:     pluginsdk.TypeString,
 							Optional: true,
 							Computed: true,
-							ValidateFunc: validation.StringInSlice([]string{
-								"512Mi",
-								"1Gi",
-								"2Gi",
-								"3Gi",
-								"4Gi",
-								"5Gi",
-								"6Gi",
-								"7Gi",
-								"8Gi",
-							}, false),
+							// There are additional options when the service team allow them
```

Comment:
```suggestion
							// NOTE: we're intentionally not validating this field since additional values are possible when enabled by the service team
```